lib/tapioca/compilers/dsl/state_machines.rb

```diff
@@ -121,16 +120,19 @@ module Dsl
       class StateMachines < Base
         extend T::Sig
 
-        sig { override.params(root: Parlour::RbiGenerator::Namespace, constant: ::StateMachines::ClassMethods).void }
+        sig { override.params(root: RBI::Tree, constant: ::StateMachines::ClassMethods).void }
         def decorate(root, constant)
           return if constant.state_machines.empty?
 
-          root.path(constant) do |klass|
+          root.create_path(T.unsafe(constant)) do |klass|
```

Comment:
`Tree#create_path` expects a `Module` but we get a `StateMachines::ClassMethods` for argument `constant` which doesn't seem to be a module.

One notable problem here is that the redefinition of `decorate` does not follow Liskov substitution principle. We do some shady thing (https://github.com/Shopify/tapioca/blob/master/lib/tapioca/compilers/dsl/base.rb#L28-L36) to avoid detection but this is weird design.

We should refactor this part to offer a correct/safe interface.